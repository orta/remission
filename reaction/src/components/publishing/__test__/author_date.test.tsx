import "jest-styled-components"
import * as React from "react"
import * as renderer from "react-test-renderer"
import AuthorDate from "../author_date"

describe("AuthorDate", () => {
  it("renders a single author", () => {
    const authors = [{ name: "Molly Gottschalk" }]
    const authorDate = renderer.create(
      <AuthorDate authors={authors} date={"2017-05-19T13:09:18.567Z"} layout={"split"} />
    )
    expect(authorDate).toMatchSnapshot()
  })

  it("renders multiple authors", () => {
    const authors = [{ name: "Molly Gottschalk" }, { name: "Kana Abe" }]
    const authorDate = renderer.create(
      <AuthorDate authors={authors} date={"2017-05-19T13:09:18.567Z"} layout={"split"} />
    )
    expect(authorDate).toMatchSnapshot()
  })
})
