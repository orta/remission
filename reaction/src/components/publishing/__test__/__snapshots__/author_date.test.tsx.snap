// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuthorDate renders a single author 1`] = `
.c1 {
  font-family: Unica77LLWebMedium ,      'Arial',      'serif';
  -webkit-font-smoothing: antialiased;
  font-size: 16px;
  line-height: 1.1em;
}

.c1.author {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: baseline;
  -webkit-box-align: baseline;
  -ms-flex-align: baseline;
  align-items: baseline;
}

.c1.author:before {
  content: "";
  display: inline-block;
  min-width: 10px;
  min-height: 10px;
  border-radius: 50%;
  margin-right: 10px;
  background-color: #000;
}

.c1.date {
  white-space: nowrap;
  margin: 0 0 0 20px;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

@media (max-width: 720px) {
  .c1 {
    font-family: Unica77LLWebMedium ,      'Arial',      'serif';
    -webkit-font-smoothing: antialiased;
    font-size: 14px;
    line-height: 1.4em;
    margin: 0 20px 0 0;
  }

  .c1.author:before {
    width: 8px;
    height: 8px;
  }
}

<div
  className="c0"
>
  <div
    className="author c1"
  >
    By 
    Molly Gottschalk
  </div>
  <div
    className="date c1"
  >
    May 19, 2017 9:09 am
  </div>
</div>
`;

exports[`AuthorDate renders multiple authors 1`] = `
.c1 {
  font-family: Unica77LLWebMedium ,      'Arial',      'serif';
  -webkit-font-smoothing: antialiased;
  font-size: 16px;
  line-height: 1.1em;
}

.c1.author {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: baseline;
  -webkit-box-align: baseline;
  -ms-flex-align: baseline;
  align-items: baseline;
}

.c1.author:before {
  content: "";
  display: inline-block;
  min-width: 10px;
  min-height: 10px;
  border-radius: 50%;
  margin-right: 10px;
  background-color: #000;
}

.c1.date {
  white-space: nowrap;
  margin: 0 0 0 20px;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

@media (max-width: 720px) {
  .c1 {
    font-family: Unica77LLWebMedium ,      'Arial',      'serif';
    -webkit-font-smoothing: antialiased;
    font-size: 14px;
    line-height: 1.4em;
    margin: 0 20px 0 0;
  }

  .c1.author:before {
    width: 8px;
    height: 8px;
  }
}

<div
  className="c0"
>
  <div
    className="author c1"
  >
    By 
    Molly Gottschalk and Kana Abe
  </div>
  <div
    className="date c1"
  >
    May 19, 2017 9:09 am
  </div>
</div>
`;
